jobs/gtfs-rt-parser/gtfs_rt_parser.py

```diff
@@ -123,109 +218,440 @@ def identify_files(glob, rt_file_type: RTFileType, progress=False) -> List[RTFil
     # its caches (at least the ones I could find)
     fs.dircache.clear()
 
-    typer.secho(f"found {len(files)} files to process", fg=typer.colors.GREEN)
+    typer.secho(
+        f"found {len(files)} {rt_file_type} files in glob {glob}", fg=typer.colors.GREEN
+    )
     return files
 
 
-# Originally this whole function was retried, but tmpdir flakiness will throw
-# exceptions in backoff's context, which ruins things
-def parse_file(bucket: str, rt_file: RTFile, pbar=None):
-    fs = get_fs()
+def download_gtfs_schedule_zip(
+    fs, gtfs_schedule_path, dst_path, pbar=None
+) -> (str, List[str]):
+    # fetch and zip gtfs schedule
+    zipname = f"{dst_path[:-1]}.zip"
+    log(
+        f"Fetching gtfs schedule data from {gtfs_schedule_path} to {zipname}",
+        pbar=pbar,
+    )
 
-    with tempfile.TemporaryDirectory() as tmp_dir:
-        parsed = parse_pb(rt_file.path, open_func=fs.open)
+    try:
+        get_with_retry(fs, gtfs_schedule_path, dst_path, recursive=True)
+    except FileNotFoundError:
+        raise ScheduleDataNotFound(f"no schedule data found for {gtfs_schedule_path}")
+
+    try:
+        os.remove(os.path.join(dst_path, "areas.txt"))
+    except FileNotFoundError:
+        pass
+
+    try:
+        os.remove(os.path.join(dst_path, "validation.json"))
+    except FileNotFoundError:
+        pass
+
+    written = []
+    with ZipFile(zipname, "w") as zf:
+        for file in os.listdir(dst_path):
+            full_path = os.path.join(dst_path, file)
+            if not os.path.isfile(full_path):
+                continue
+            with open(full_path) as f:
+                zf.writestr(file, f.read())
+            written.append(file)
+
+    if not written:
+        raise ScheduleDataNotFound(f"no schedule data found for {gtfs_schedule_path}")
+
+    return zipname, written
+
+
+def execute_rt_validator(
+    gtfs_file: str, rt_path: str, jar_path: Path, verbose=False, pbar=None
+):
+    log(f"validating {rt_path} with {gtfs_file}", fg=typer.colors.MAGENTA, pbar=pbar)
+
+    args = [
+        "java",
+        "-jar",
+        str(jar_path),
+        "-gtfs",
+        gtfs_file,
+        "-gtfsRealtimePath",
+        rt_path,
+        "-sort",
+        "name",
+    ]
+
+    log(f"executing rt_validator: {' '.join(args)}", pbar=pbar)
+    subprocess.run(
+        args,
+        capture_output=True,
+    ).check_returncode()
+
+
+def validate_and_upload(
+    fs,
+    jar_path: Path,
+    dst_path_gtfs,
+    dst_path_rt,
+    tmp_dir,
+    hour: RTAggregation,
```

Comment:
**nitpick:** can we not call this "hour" but instead "aggregation"? to make it generic in case we change granularity?